File: src/main/java/io/github/kingg22/api_vacunas_panama/web/dto/IdNombreDto.java
package io.github.kingg22.api_vacunas_panama.web.dto;

import java.io.Serializable;

public record IdNombreDto(Long id, String nombre) implements Serializable {
    public IdNombreDto(Short id, String nombre) {
        this(id.longValue(), nombre);
    }

    public IdNombreDto(Integer id, String nombre) {
        this(id.longValue(), nombre);
    }
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/web/dto/IdNombreDto.kt
package io.github.kingg22.api.vacunas.panama.web.dto

import java.io.Serializable

@JvmRecord
data class IdNombreDto(val id: Long, val nombre: String?) : Serializable {
    constructor(id: Short, nombre: String?) : this(id.toLong(), nombre)
    constructor(id: Int, nombre: String?) : this(id.toLong(), nombre)
}
